### DISCUSSION TITLE
Cannot Delete Device, only via dropdown

### DISCUSSION THREAD

**Discussion by NollieTre** [netbox-community/netbox#10872] [Q&A] [answered]
Hey i have upgraded Netbox from 3.0.8 -> 3.3.7 yesterday and it appears that i cannot delete devices from the respective device menu itself, only in the dropdown. 
![Bildschirmfoto 2022-11-09 um 13 26 21](https://user-images.githubusercontent.com/117828680/200830525-0b57dcb2-b985-4fe7-8bf3-c2626535b721.png) this works.

![Bildschirmfoto 2022-11-09 um 13 26 34](https://user-images.githubusercontent.com/117828680/200830961-2144fbfe-de0c-4368-a0f4-2b9d2b670556.png) and this doesnt work, it shows me the following:

![Bildschirmfoto 2022-11-09 um 13 27 11](https://user-images.githubusercontent.com/117828680/200830636-a3ac4c4b-87ee-42dc-876c-5a6998131b74.png)
images.githubusercontent.com/117828680/200830625-338c2ffa-8ebf-4937-9c88-b8e85b18662d.png)

when clicking the delete button inside the device options it just opens an empty field.
What are my options here? is there a fix or something i am missing?

**Comment by candlerb** [comment, marked as answer]
I presume you're looking at `/dcim/devices/1159/`.  This will show you the device details page.

When you click Delete, this page should remain displayed but dimmed, and a pop-up should appear in front of it:

![image](https://user-images.githubusercontent.com/44789/201008061-0abced85-850a-45c3-8922-d1cdb4ab55bb.png)

This works for me (just tested it with 3.3.7 and Chrome), so something is wrong with your system.  Maybe you are blocking Javascript?  Maybe your browser has some old Javascript cached? (Try clicking shift-reload then try again).  Maybe the upgrade process missed the `collectstatic` step?  (But if you ran `./upgrade.sh` that should have been taken care of)

Since Javascript is involved, I suggest you open your browser's developer console, *then* click the Delete button, and see if you get any Javascript errors logged.

**Reply by NollieTre** [reply]
Yes so, i have it running on a localhost, where it works now after clearing my cache and re-enabling javascript. for some reason it does not work on the live website.. but thank you for the answer, maybe i am missing something in plain sight! 
helped me alot and i guess the answer is in here.
